Specify stock rates per location type

esoergel opened the pull request.

esoergel committed:
Use real location types in stock levels view

esoergel committed:
[WIP] Use sql location types' stock_levels

Commit:
WIP

esoergel committed:
rename migrations after rebasing in new one

esoergel committed:
Make stock_category depend on location type

esoergel committed:
Skip domains without commtrack configs

Commit:
refactoring

Commit:
refactoring part2

Commit:
added Arrived Tests

Commit:
Added tests for sms handlers

Commit:
Merge branch 'master' into ews-sms-scripts2

Commit:
Style tweaks

Commit:
Added translations, removed commented code

Commit:
Removed unnecessary line

Commit:
added tests for handlers in ILS

Commit:
flake

Commit:
flake

esoergel committed:
Document FormListForm usage

esoergel committed:
Validate that emergency < low stock < overstock

dimagimon commented on corehq/apps/reports/commtrack/data_sources.py:
```diff
@@ -322,11 +324,7 @@ def _convert_to_daily(consumption):
-                            _convert_to_daily(consumption),
-                            Domain.get_by_name(self.domain)
-                        ),
+                        'category': state_stock_category(state),
```
F821 undefined name 'state_stock_category'

dimagimon commented on corehq/ex-submodules/casexml/apps/stock/utils.py:
```diff
@@ -20,24 +20,23 @@ def stock_category(stock, daily_consumption, domain):
     if months_left is None:
         return 'nodata'
-    elif months_left < stock_levels.understock_threshold:
+    elif months_left < understock_threshold:
```
F821 undefined name 'understock_threshold'

dimagimon commented on corehq/ex-submodules/casexml/apps/stock/utils.py:
```diff
@@ -20,24 +20,23 @@ def stock_category(stock, daily_consumption, domain):
+    elif months_left < understock_threshold:
         return 'understock'
-    elif months_left > stock_levels.overstock_threshold:
+    elif months_left > overstock_threshold:
```
F821 undefined name 'overstock_threshold'

benrudolph commented on corehq/apps/commtrack/views.py:
```diff
@@ -111,6 +111,9 @@ def post(self, request, *args, **kwargs):
                     setattr(self.commtrack_settings.stock_levels_config, field,
                             data['stock_' + field])
+            # touch the location_types to update their stock levels config
+            for loc_type in LocationType.objects.filter(domain=self.domain).all():
```
rails has a nice utility for this called `touch`.  that does exactly what your comment suggests. maybe it'll be added in django 2.0

benrudolph commented on corehq/apps/hqwebapp/templates/hqwebapp/partials/form_list_form.html:
```diff
@@ -25,7 +25,7 @@
 
         <tbody data-bind="foreach: rows">
-            <tr>
+            <tr data-bind="css: { error: Object.keys(errors).length !== 0 }">
```
Usually better to use `_.keys` than `Object.keys` since underscore will polyfill for you if the browser doesn't have `Object.keys`

esoergel replied:
Good to know, thanks!  I was shocked to find out that JS doesn't seem to have any good way of doing `my_hashmap.isEmpty;`, `my_hashmap === {}`, or `my_hashmap.length === 0;`

benrudolph replied:
yea it's pretty bad. don't know why i didn't mention this before, but underscore has a method for just (that)[http://underscorejs.org/#isEmpty] `_.isEmpty`

esoergel committed:
fix bad variable names

esoergel committed:
Add fake sms util for testing

esoergel committed:
PR feedback

czue commented on corehq/apps/commtrack/views.py:
```diff
@@ -111,6 +111,9 @@ def post(self, request, *args, **kwargs):
                             data['stock_' + field])
+            # touch the location_types to update their stock levels config
+            for loc_type in LocationType.objects.filter(domain=self.domain).all():
+                loc_type.save()
```
i would make this more explicit

czue commented on corehq/apps/commtrack/sms.py:
```diff
@@ -62,6 +62,19 @@ def handle(verified_contact, text, msg=None):
     return True
 
 
+def fake_sms(user, text):
```
might want to make it more clear this is a test util (maybe move to tests or similar)

esoergel committed:
Misc minor cleanups

Commit:
Merge branch 'master' into ews-sms-scripts

Conflicts:
	custom/ewsghana/alerts/alerts.py
	custom/ewsghana/utils.py

dannyroberts committed:
remove unused phone templates and templatetag

NoahCarnahan committed:
Second attempt at creating auto expanding column

NoahCarnahan committed:
First stab at auto expanding column

Commit:
Merge branch 'master' into ews-sms-scripts

Conflicts:
	custom/ewsghana/alerts/alerts.py
	custom/ewsghana/utils.py

Commit:
Merge branch 'master' into ews-sms-scripts

Conflicts:
	custom/ilsgateway/tanzania/reminders/__init__.py

NoahCarnahan committed:
Move function

NoahCarnahan committed:
Remove unused filter parameters

NoahCarnahan committed:
Reference column config field directly

NoahCarnahan committed:
Revert "First stab at auto expanding column"

This reverts commit 9e089ad00e18670c97247f030431727ffbe2ac6a.

NoahCarnahan committed:
Use column_config.report_column_id instead of field for data slug and sumwhen alias

NoahCarnahan committed:
Get the table more efficiently

NoahCarnahan committed:
Wrap session usage in try/catch

NoahCarnahan committed:
Close the session

NoahCarnahan committed:
Don't fail if first column is an expand

NoahCarnahan committed:
Limit expansion to a maximum width of 10

NoahCarnahan committed:
Show column warnings to user

NoahCarnahan committed:
Don't error if the table doesn't exist yet

NoahCarnahan committed:
Reload the page when the data table is built so that the datatable columns will be updated

Commit:
Added tests to EWSGhana module

NoahCarnahan committed:
Fix whitespace

NoahCarnahan committed:
Merge remote-tracking branch 'origin/master' into expand-column-2

NoahCarnahan committed:
Merge remote-tracking branch 'origin/master' into expand-column-2

Conflicts:
	corehq/apps/userreports/reports/view.py

NoahCarnahan committed:
break get_expanded_columns into smaller chunks.

kaapstorm committed:
Import org unit and convert gender value

Commit:
Merge branch 'master' into logistics-tests

Commit:
Changed facility manager test

Commit:
Small change in product sync

Commit:
changes in Patient Info Report

Commit:
Added creating roles in tests

NoahCarnahan committed:
Add simple test for column expansion

Commit:
Fixed tests

agrzywinski committed:
EWS Edits: Reporting Rates Report

Commit:
Tweak in runner

Commit:
Changes in Patient Interactions report

snopoke committed:
translate and filter form summary

snopoke committed:
allow displaying labels for question ids

snopoke committed:
show calculations

snopoke committed:
show relevance and required

snopoke committed:
set folder icon colors

snopoke committed:
add required and relevant to test data

Commit:
refactored code

Commit:
flake8

czue committed:
Merge pull request #5969 from dimagi/phone-templates

remove unused phone templates and templatetag

dannyroberts committed:
simplify sms languages' use of app langs template

(tiny/cleanup) use {% include ... with ... %} rather than passing in all the way from
the view context

czue committed:
Merge pull request #5968 from lwyszomi/ucla_reports

Ucla reports

czue committed:
Merge pull request #5964 from lwyszomi/155959

EWS Edits: Reporting Rates Report

czue committed:
Merge pull request #5744 from lwyszomi/ews-sms-scripts

SMS test scripts for EWS and ILS

Commit:
Refactored get test locations function

czue committed:
Merge pull request #5966 from lwyszomi/runner-test-facilities

ILSGateway: Tweak in runner

czue committed:
Merge pull request #5511 from dimagi/loc-type-stock-rates

Move LocationType to SQL

TylerSheffels committed:
Updating submodules

benrudolph committed:
Ignore performance files

czue committed:
fix bad import

dannyroberts committed:
Merge pull request #5973 from dimagi/hotfix

fix bad import

esoergel committed:
Update ewsghana tests to use SQL LocationTypes

czue committed:
make tests more explicit about what they depend on

esoergel committed:
Update ilsgateway tests to use SQL LocationType

czue committed:
Merge pull request #5975 from dimagi/custom-tests

Update custom reports tests to use SQL Location Type

czue committed:
pull out domain

esoergel committed:
Merge pull request #5972 from benrudolph/app-builder-live-gitignore

Ignore performance files

esoergel committed:
Merge pull request #5971 from dimagi/sms-langs-hide-deploy-column

simplify sms languages' use of app langs template

czue committed:
reorder migrations

czue committed:
manually update migrations to reflect reordering

czue committed:
properly default data to False

czue committed:
fix 500 i got locally

esoergel committed:
Merge pull request #5976 from dimagi/location-migration-hotfix

Location migration hotfix

czue committed:
fix test

dannyroberts committed:
allow people w/ Developer toggle to access admin APIs

czue committed:
Merge pull request #5977 from dimagi/developer-admin-apis

allow people w/ Developer toggle to access admin APIs

dannyroberts committed:
bump jsonobject to 0.6.3

czue committed:
Merge pull request #5978 from dimagi/jsonobject-0.6.3

bump jsonobject to 0.6.3

snopoke committed:
Merge pull request #5967 from dimagi/form-summary

UAT changes to form summary

Commit:
Use couch_location as property not method

Commit:
Fixed location type bugs in logistics modules

Commit:
Lint errors

Commit:
Merge branch 'master' into logistics-tests

Conflicts:
	custom/ewsghana/tests/__init__.py

czue committed:
Merge pull request #5979 from lwyszomi/logistics-couch-location-property

Use couch_location as property not method

czue committed:
Merge pull request #5980 from lwyszomi/logistics-location-types

Fixed location type bugs in logistics modules

czue committed:
Merge pull request #5981 from lwyszomi/logistics-tests

Added tests to ewsghana module

NoahCarnahan committed:
Add test for retreiving distinct values

Commit:
Added check for old domains

czue committed:
Merge pull request #5982 from lwyszomi/logistics-legacy

Logistics: Added check for old domains

NoahCarnahan committed:
Add test for too many distinct values and no data

NoahCarnahan committed:
Fix lint

snopoke committed:
Merge pull request #5913 from dimagi/expand-column-2

Add "auto-expanding" column type to UCR

benrudolph committed:
Merge pull request #5974 from dimagi/explicit-tests

make tests more explicit about what they depend on

Commit:
Deleting existing location types, removed not needed method

NoahCarnahan committed:
Update readme

czue committed:
Merge pull request #5984 from dimagi/expand-column-2

Add column expansion to userreports readme

czue committed:
Merge pull request #5983 from lwyszomi/location-type-fix-again

Logistics: another location type tweak

czue committed:
Merge pull request #5962 from dimagi/dhis2_fetch_cases

Import org unit and convert gender value

sravfeyn committed:
update staging.yaml

[ci skip]

NoahCarnahan committed:
Fix column expansion bug

NoahCarnahan committed:
add test

snopoke committed:
use asterisk to mark required questions

snopoke committed:
Merge pull request #5987 from dimagi/expand-column-2

UCR "expand column" bugfix

NoahCarnahan committed:
Merge pull request #5988 from dimagi/required

use asterisk to mark required questions

snopoke committed:
update hidden value icon

NoahCarnahan committed:
Merge pull request #5991 from dimagi/variable

update hidden value icon

esoergel committed:
Misc cleanup
